arkfilepicker/src/main/java/space/taran/arkfilepicker/folders/FoldersRepo.kt

```diff
@@ -119,11 +113,57 @@ class FoldersRepo(private val appCtx: Context) {
         } ?: listOf(fav)
         folders = mutFolders
 
-        val arkFolder = root.arkFolder()
-        require(arkFolder.exists()) { "Ark folder must exist" }
-        val favsFile = arkFolder.arkFavorites()
-        val jsonFavs = JsonFavorites(folders[root]!!.map(Path::toString))
-        favsFile.writeText(klaxon.toJsonString(jsonFavs))
+        writeFavorites(root)
+    }
+
+    suspend fun forgetRoot(root: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                val mFolders = folders.toMutableMap()
+                mFolders.remove(root)
+                folders = mFolders
```

Comment:
Probably we can just perform an immutable deletion.

Reply:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/minus.html

Reply:
Alright, cool

Reply:
Immutable deletion using ```minus``` is not working, so I reverted to mutable removal until we find a better way to perform immutable removal

Reply:
Immutable deletion using ```minus``` is not working, so I reverted to mutable removal until we find a better way to perform immutable removal